en/sensor/rangefinders.md

```diff
@@ -0,0 +1,183 @@
+
+TeraRanger provide a number of lightweight distance measurement sensor based on infrared Time-of-Flight (ToF) technology. They are typically faster and have greater range than sonar, and smaller and lighter than laser-based systems. PX4 supports:
+* [TeraRanger One](http://www.teraranger.com/products/teraranger-one/) (0.2 - 14 m)
+* [TeraRanger Evo](https://www.terabee.com/portfolio-item/teraranger-evo-infrared-distance-sensor/) (0.5 – 60 m)
+
+These rangefinders must be connected via the I2C bus (using an [I2C adapter](http://www.teraranger.com/product/teraranger-i2c-adapter/)).
+
+The sensors are enabled using the parameter [SENS_EN_TRANGER](../advanced_config/parameter_reference.md#SENS_EN_TRANGER) (you can set the type of sensor or that PX4 should auto-detect the type).
+
+> **Info** The *Terranger One* is used in the [Qualcomm Snapdragon Flight](../flight_controller/snapdragon_flight.md). 
+
+### uLanding Radar
+
+The *Aerotenna* [uLanding](https://aerotenna.com/sensors/#ulanding) altimeter is compact microwave rangefinder that has been optimised for use on UAVs. It has a sensing range of 45m. A particular advantages of this product are that it can operate effectively in all weather conditions and over all terrain types (including water).
+
+The uLanding Radar is not present in "most" firmware by default and must be started by updating a configuration file (rather than by a parameter). More information can be found here: [uLanding Radar](../sensor/ulanding_radar.md).
+
+
+### Other
+
+PX4 also supports the Bebop rangefinder.
+
+
+## Rangefinder Configuration
+
+### Setup
+
+The rangefinder is configured using [EKF2\_RNG\_*](../advanced_config/parameter_reference.md#EKF2_RNG_POS_X) parameters. These configure information about the offset of the rangefinder from the centre of the vehicle body, the approximate delay of data reaching the estimator from the sensor, etc.
+
+### Enable/Use
+
+The rangefinder can be enabled in two ways:
+1. Set [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE) to *Range finder* (`2`). This makes the rangefinder the primary source of height estimation (the default altitude sensor is the barometer).
+1. Set [EKF2_RNG_AID](../advanced_config/parameter_reference.md#EKF2_RNG_AID) to `1`. This makes the vehicle use the rangefinder as the primary source when it is safe to use, but will otherwise use the sensor specified in `EKF2_HGT_MODE`.
+   * Specifically, the rangefinder is enabled when: 
+     * velocity < [EKF2_RNG_A_VMAX](../advanced_config/parameter_reference.md#EKF2_RNG_A_VMAX)
+     * distance to ground < [EKF2_RNG_A_HMAX](../advanced_config/parameter_reference.md#EKF2_RNG_A_HMAX)
+   * Other parameters affecting "Range aid" are prefixed with `EKF2\_RNG\_A\_`.
+
+## Testing
+
+The easiest way to test the rangefinder is to vary the range and compare to the values detected by PX4. The sections below show some approaches to getting the measured range.
+
+### QGroundControl Analyze Tool 

```

Comment:
Thanks @bkueng @ChristophTobler  - using wrong vehicle was the cause. Tested against the gazebo simulation and things worked well.  This section now updated.

Anything else needed? I think this is a much more complete rangefinder story than we had.